boa_engine/src/bytecompiler/class.rs

```diff
@@ -138,6 +138,7 @@ impl ByteCompiler<'_, '_> {
         let index = self.code_block.functions.len() as u32;
         self.code_block.functions.push(code);
         self.emit(Opcode::GetFunction, &[index]);
+        self.emit_u8(0);
```

Comment:
Thoughts on creating an `emit` method like `emit_with_opcode_and_u8` (maybe not the best name) to bind the pushing of the two operands together?

Reply:
I think we could maybe generate `emit` functions for every opcode via a macro e.g. `Opcode::GetFunction::emit`. These functions could then automatically take the correct arguments.
Do we want to create a dedicated issue for making bytecode generation more typesafe?

Reply:
Yeah, that would make sense to me.